kernel/syscall.c

```diff
@@ -2671,8 +2671,26 @@ long myst_syscall_getsockname(
     myst_fdtable_t* fdtable = myst_fdtable_current();
     myst_sockdev_t* sd;
     myst_sock_t* sock;
+    myst_fdtable_type_t type;
+    void* device = NULL;
+    void* object = NULL;
+
+    if (!myst_is_good_addr(addrlen) || !myst_is_good_addr(addr))
+    {
+        ret = -EFAULT;
```

Comment:
I'd replace these two lines with:

```C++
ERAISE(-EFAULT);
```

Reply:
Oh, Ming-Wei already commented on that. :)

Reply:
Added, thank you!